DBAL/TracingConnectionFactory.php

```diff
@@ -43,17 +43,7 @@ public function createConnection(
         array $mappingTypes = []
     ): Connection {
         $connection = $this->connectionFactory->createConnection($params, $config, $eventManager, $mappingTypes);
-        $driverConnection = new TracingDriverConnection(
-            $connection->getWrappedConnection(),
-            $this->tracing,
-            $this->spanFactory,
-            $connection->getUsername()
-        );
-        $reflectionObject = new ReflectionObject($connection);
-        $property = $reflectionObject->getProperty('_conn');
-        $property->setAccessible(true);
-        $property->setValue($connection, $driverConnection);
-        $property->setAccessible(false);
+        $connection->getEventManager()->addEventListener(Events::postConnect, $this->eventListener);
```

Comment:
Not directly connected to this line of code, but it would be great to have a functional test case covering the source problem which lead to this PR. If you need some help with it (the process is optimized for reusability on travis, not developers), feel free to contact me.

Reply:
I think I got it, it starts a docker container running jaeger, then stops it.

Functional tests start/stop the container in setup/teardown.

What I don’t know is how to test that the connection is not initiated ?

I understand all the plumbing involved but fail to see what should be written in the test.

Reply:
As written, I will provide a functional test case after this PR is merged. Closing this conversation for now.